core/runtime/src/main/java/io/quarkiverse/langchain4j/QuarkusAiServicesFactory.java

```diff
@@ -54,6 +58,29 @@ public AiServices<T> tools(List<Object> objectsWithTools) {
             return this;
         }
 
+        List<ToolMethodCreateInfo> lookup(String className) {
+            Map<String, List<ToolMethodCreateInfo>> metadata = ToolsRecorder.getMetadata();
+            var fast = metadata.get(className);
+            if (fast != null) {
+                return fast;
+            }
+
+            for (Map.Entry<String, List<ToolMethodCreateInfo>> entry : metadata.entrySet()) {
+                if (entry.getKey().endsWith(className)) {
+                    return entry.getValue();
+                }
+                if (className.endsWith("_ClientProxy")) { // Arc specific
```

Comment:
I think it would better to use: `io.quarkus.arc.ClientProxy#unwrap`

Reply:
Yes, much simpler. I was unsure if it would support the nested classes.